core/src/main/java/hudson/model/Queue.java

```diff
@@ -209,7 +209,7 @@
      *
      * This map is forgetful, since we can't remember everything that executed in the past.
      */
-    private final Cache<Long,LeftItem> leftItems = CacheBuilder.newBuilder().expireAfterWrite(5*60, TimeUnit.SECONDS).build();
+    private final Cache<Long,LeftItem> leftItems = CacheBuilder.newBuilder().maximumSize(200).expireAfterWrite(5*60, TimeUnit.SECONDS).build();
```

Comment:
Minor thing: if we are touching this line, maybe this is a best time to replace `5*60, TimeUnit.SECONDS` by `5, TimeUnit.MINUTES`?

Reply:
@Jimilian A good point, and I'm going to limit it a touch more because it feels like we're carrying around still a bit too much baggage here.